src/plugins/share/public/components/share_tabs.tsx

```diff
@@ -34,7 +34,10 @@ export const ShareMenuTabs = () => {
 
   tabs.push(linkTab);
 
-  if (shareMenuItems.length > 0) {
+  const enabledItems = shareMenuItems.filter(({ shareMenuItem }) => !shareMenuItem?.disabled);
+
+  // do not show the export tab if the license is disabled
+  if (enabledItems.length > 0) {
```

Comment:
It looks like https://github.com/elastic/kibana/pull/188439 was not backported to 8.15. In order to have the unit tests for this file, I'm including parts of that fix into this backport.